File: android/src/main/java/expo/modules/universaltooltip/records/TextStyle.kt
package expo.modules.universaltooltip.records
import android.graphics.Typeface
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

data class TextStyle(
    @Field var fontSize: Float = 13f,
    @Field var color: Int = -16777216,
    @Field var fontFamily: String?,
    @Field var fontWeight: String = "normal",
) : Record {

}

fun convertFontWeightToTypeface(fontWeight: String): Typeface {
    return when (fontWeight) {
        "normal" -> Typeface.create(Typeface.DEFAULT, Typeface.NORMAL)
        "bold" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        "100" -> Typeface.create(Typeface.DEFAULT, Typeface.NORMAL)
        "200" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        "300" -> Typeface.create(Typeface.DEFAULT, Typeface.NORMAL)
        "400" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        "500" -> Typeface.create(Typeface.DEFAULT, Typeface.NORMAL)
        "600" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        "700" -> Typeface.create(Typeface.DEFAULT, Typeface.NORMAL)
        "800" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        "900" -> Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
        else -> Typeface.DEFAULT
    }
}

File: ios/TextStyle.swift
//
//  TextStyle.swift
//  UniversalTooltip
//
//  Created by Alan Toa on 2022/12/31.
//
import ExpoModulesCore
import SwiftUI

struct TextStyle: Record {
  @Field
  var fontSize: Double = 13
  
  @Field
  var color: UIColor = .black
  
  @Field
  var fontFamily: String?
  
  @Field
  var fontWeight: String = "normal"
}

func fontWeightToSwiftUI(_ weight: String) -> Font.Weight {
  switch weight {
    case "normal", "400":
      return .regular
    case "bold", "700":
      return .bold
    case "100":
      return .ultraLight
    case "200":
      return .thin
    case "300":
      return .light
    case "500":
      return .medium
    case "600":
      return .semibold
    case "800":
      return .heavy
    case "900":
      return .black
    default:
      return .regular
  }
}
